compiler/il/OMRILOps.hpp

```diff
@@ -117,21 +87,53 @@ class ILOpCode
    *  \param operation
    *     Vector operation
    *
-   *  \param resultVectorType
-   *     Type of the vector created by the operation
+   *  \param vectorType
+   *     Type of the vector which is either source or result (or both)
    *
    *  \return
    *     Vector opcode
    */
-   static TR::ILOpCodes createVectorOpCode(VectorOperation operation, DataType resultVectorType)
+   static TR::ILOpCodes createVectorOpCode(TR::VectorOperation operation, DataType vectorType)
       {
-      static_assert(NumAllIlOps < ((uint64_t)1 << (sizeof(TR::ILOpCodes)*8)), "Number of scalar and vector opcodes does not fit into TR::ILOpCodes\n");
+      TR_ASSERT_FATAL(vectorType.isVector(), "createVectorOpCode should take vector type\n");
 
-      TR_ASSERT_FATAL(resultVectorType.isVector(), "createVectorOpCode should take vector type\n");
+      TR_ASSERT_FATAL(operation < TR::firstTwoVectorTypeOperation, "Vector operation should be one vector type operation\n");
 
-      return (TR::ILOpCodes)(TR::NumScalarIlOps + operation*TR::NumVectorTypes + (resultVectorType - TR::NumScalarTypes));
+      return (TR::ILOpCodes)(TR::NumScalarIlOps + operation*TR::NumVectorTypes + (vectorType - TR::NumScalarTypes));
       }
 
+  /** \brief
+   *     Creates vector opcode given vector operation and resulting vector type
+   *
+   *  \param operation
+   *     Vector operation
+   *
+   *  \param vectorType
+   *     Type of the vector which is either source or result (or both)
+   *
+   *  \return
+   *     Vector opcode
+   */
+   static TR::ILOpCodes createVectorOpCode(TR::VectorOperation operation, DataType srcVectorType, DataType resVectorType)
+      {
+      TR_ASSERT_FATAL(srcVectorType.isVector(), "createVectorOpCode should take vector source type\n");
+      TR_ASSERT_FATAL(resVectorType.isVector(), "createVectorOpCode should take vector result type\n");
+
+      TR_ASSERT_FATAL(operation >= TR::firstTwoVectorTypeOperation, "Vector operation should be two vector type operation\n");
+
+      return (TR::ILOpCodes)(TR::NumScalarIlOps + TR::NumOneVectorTypeOps +
+                             operation * TR::NumVectorTypes * TR::NumVectorTypes +
+                             (srcVectorType - TR::NumScalarTypes) * TR::NumVectorTypes +
+                             (resVectorType - TR::NumScalarTypes));
+      }
+
+   bool isTwoTypeVectorOpCode()
+         {
+         if (!isVectorOpCode()) return false;
+
+         return (getVectorOperation() >= TR::firstTwoVectorTypeOperation);
```

Comment:
Can you make the naming consistent with the thing you're checking?  i.e., is it "TwoTypeVector" or "TwoVectorType"

Reply:
Done: TwoTypeVector